domain/src/test/java/org/oppia/domain/topic/StoryProgressControllerTest.kt

```diff
@@ -517,7 +514,8 @@ class StoryProgressControllerTest {
 
   private fun verifyProfileProgressFailed() {
     verify(mockProfileProgressObserver, atLeastOnce()).onChanged(profileProgressResultCaptor.capture())
-    assertThat(profileProgressResultCaptor.value.isFailure()).isTrue()
+    val topicProgressDatabase = profileProgressResultCaptor.value.getOrThrow()
+    assertThat(topicProgressDatabase.topicProgressMap.size).isEqualTo(0)
```

Comment:
Could you assert that it's empty, instead? E.g.:

```kotlin
assertThat(topicProgressDatabase.topicProgressMap).isEmpty()
```

Ditto below.

Reply:
Updated this to check failure instead
```
verify(mockTopicProgressObserver, atLeastOnce()).onChanged(topicProgressResultCaptor.capture())
assertThat(topicProgressResultCaptor.value.isFailure()).isTrue()
```